pymc/step_methods/external/nutpie.py

```diff
@@ -0,0 +1,242 @@
+#   Copyright 2024 - present The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import logging
+import warnings
+
+from typing import Literal
+
+from arviz import InferenceData
+
+from pymc.backends.arviz import coords_and_dims_for_inferencedata, find_constants, find_observations
+from pymc.model import Model
+from pymc.step_methods.compound import Competence
+from pymc.step_methods.external.base import ExternalSampler
+from pymc.vartypes import continuous_types
+
+logger = logging.getLogger("pymc")
+
+try:
+    import nutpie
```

Comment:
nutpie should only be imported if the step method is used/instantiated, so as to avoid import time penalty